packages/service-clients/odsp-client/src/odspClient.ts

```diff
@@ -146,53 +149,42 @@ export class OdspClient {
 		container: IContainer,
 		connection: OdspConnectionConfig,
 	): Promise<IFluidContainer> {
-		const createNewRequest: IRequest = createOdspCreateContainerRequest(
-			connection.siteUrl,
-			connection.driveId,
-			connection.folderPath,
-			uuid(),
-		);
-
 		// eslint-disable-next-line import/no-deprecated
 		const rootDataObject = await requestFluidObject<IRootDataObject>(container, "/");
 
 		/**
 		 * See {@link FluidContainer.attach}
 		 */
 		const attach = async (): Promise<string> => {
+			const createNewRequest: IRequest = createOdspCreateContainerRequest(
+				connection.siteUrl,
+				connection.driveId,
+				"",
+				uuid(),
+			);
 			if (container.attachState !== AttachState.Detached) {
 				throw new Error("Cannot attach container. Container is not in detached state");
 			}
-			await container.attach(createNewRequest);
-			const absoluteUrl = await container.getAbsoluteUrl("/");
-			if (absoluteUrl === undefined) {
-				throw new Error("Absolute Url not avaiable on attached container");
+			const resolvedUrl = container.resolvedUrl as IOdspResolvedUrl;
```

Comment:
use getOdspResolvedUrl

Reply:
we should also never use `as` casting

Reply:
can `getOdspResolvedUrl` be marked as public? api extractor complains that there is no release tag added?

Reply:
i see. a better change might be to add a new function that is actually a type guard
``` typescript
function isOdspResolveUrl(maybe: IResolvedUrl): maybe is IOdspResolvedUrl{
return "odspResolvedUrl" in maybe;
}
```
export that function, and use it in the existing function, as you probably don't want an assert on failure